Github workflow failure

Today, I didn't receive the email and the job had failed. I looked into the issue and had to manually update the `pnpm-lock.yaml` file. Not sure if this is the right approach and we are supposed to update the lock file once in a while.

` ERR_PNPM_FROZEN_LOCKFILE_WITH_OUTDATED_LOCKFILE  Cannot perform a frozen installation because the lockfile needs updates`

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -3,7 +3,7 @@ name: Scheduled Email
 on:
   workflow_dispatch: {}
   schedule:
-#   Runs at 04:00 UTC every day.
+#   Eg: Runs at 04:00 UTC every day.
     - cron: '0 4 * * *'
   
 jobs:
@@ -15,17 +15,19 @@ jobs:
 
     steps:
       - uses: actions/checkout@v3
-      
+      - uses: pnpm/action-setup@v2
+        with:
+          version: 7
+
       - name: Node LTS versions
         uses: actions/setup-node@v3
-          
         with:
           node-version: ${{ matrix.node-version }}
+          cache: 'pnpm'
           
       - name: Install Node.js dependencies
         run: |
-          npm i -g pnpm
-          pnpm i
+          pnpm i --no-frozen-lockfile
         
       - env:
             SENDER_EMAIL: ${{ secrets.SENDER_EMAIL }}

--- a/package.json
+++ b/package.json
@@ -4,7 +4,7 @@
   "description": "",
   "main": "src/index.ts",
   "scripts": {
-    "start": "ts-node src/index.ts"
+    "start": "TS_NODE_FILES=true TS_NODE_TRANSPILE_ONLY=true ts-node src/index.ts"
   },
   "keywords": [],
   "author": "tnmyk",

